src/network/networkctl.c

```diff
@@ -3192,6 +3192,118 @@ static int verb_cat(int argc, char *argv[], void *userdata) {
         return ret;
 }
 
+static int get_config_files_by_link(
+                sd_bus *bus,
+                const char *link,
+                char **ret_path,
+                char ***ret_dropins) {
+
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_strv_free_ char **dropins = NULL;
+        _cleanup_free_ char *path = NULL;
+        int r, ifindex;
+
+        assert(bus);
+        assert(link);
+        assert(ret_path);
+        assert(ret_dropins);
+
+        r = bus_call_method(bus, bus_network_mgr, "GetLinkByName", &error, &reply, "s", link);
```

Comment:
you use sd_network_link_get_network_file() below, which bypasses the bus. It's a bit weird to first go via the bus and then bypass it. you might as well bypass it entirely then, resolving link names to ifindexes is easily done via rtnl_resolve_interface() or rtnl_resolve_interface_or_warn()